app/components/MusicPlayerBar.tsx

```diff
@@ -0,0 +1,255 @@
+
+  const playlist = usePlayList();
+  const videoInfo = useMemo(() => playlist.getCurrentMusic(), [playlist]);
+
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [player, setPlayer] = useState<YouTubePlayer>();
+  const [currentTime, setCurrentTime] = useState(0);
+  const [duration, setDuration] = useState(0);
+  const [volume, setVolume] = useState(100);
+  const [isMuted, setIsMuted] = useState(false);
+  const [repeatMode, setRepeatMode] = useState<'NO_REPEAT' | 'REPEAT_ONE' | 'REPEAT_ALL'>('NO_REPEAT');
+  const [isOpen, setOpen] = useState(false);
+  const [shuffle, setShuffle] = useState(false);
+
+  const [timer, setTimer] = useState<any>();
+
+  const percent = useMemo(() => (currentTime / duration) * 100, [currentTime, duration]);
+
+  const PlayOrPause: SvgIconComponent = useMemo(() => isPlaying ? Pause : PlayArrow, [isPlaying]);
+
+  const VolumeIcon: SvgIconComponent = useMemo(() => {
+    if (isMuted) return VolumeOff;
+    if (volume === 0) return VolumeMute;
+    else if (volume < 50) return VolumeDown;
+    else return VolumeUp;
+  }, [isMuted, volume]);
+
+  const RepeatIcon: SvgIconComponent = useMemo(() => repeatMode === 'REPEAT_ONE' ? RepeatOne : Repeat, [repeatMode]);
+
+  const onReady = useCallback((event: YouTubeEvent) => {
+    setPlayer(event.target);
+    event.target?.playVideo();
+    setIsPlaying(true);
+    setDuration(event.target?.getDuration());
+
+    event.target?.setVolume(volume);
+    if (isMuted) {
+      event.target?.mute();
+    } else {
+      event.target?.unMute();
+    }
+
+    clearInterval(timer);
+    setTimer(setInterval(() => {
+      setCurrentTime(event.target?.getCurrentTime() ?? 0);
+    }, 100));
+  }, [isMuted, volume, timer]);
+
+  const onEnd = useCallback(() => {
+    if (repeatMode === 'NO_REPEAT') {
+      playlist.next();
+    } else if (repeatMode === 'REPEAT_ONE') {
+      player?.playVideo();
+    } else {
+      playlist.next(true);
+    }
+  }, [player, playlist, repeatMode]);
+
+  const onTogglePlay = useCallback(() => {
+    if (isPlaying) {
+      player?.pauseVideo();
+    } else {
+      player?.playVideo();
+    }
+    setIsPlaying(!isPlaying);
+  }, [isPlaying, player]);
+  const onToggleMute = useCallback(() => {
+    setIsMuted(!isMuted);
+    if (isMuted) {
+      player?.unMute();
+    } else {
+      player?.mute();
+    }
+  }, [player, isMuted]);
+  const onToggleRepeat = useCallback(() => {
+    if (repeatMode === 'NO_REPEAT') setRepeatMode('REPEAT_ALL');
+    else if (repeatMode === 'REPEAT_ALL') setRepeatMode('REPEAT_ONE');
+    else setRepeatMode('NO_REPEAT');
+  }, [repeatMode]);
+  const onToggleOpen = useCallback(() => setOpen(!isOpen), [isOpen]);
+
+  const onShuffle = useCallback(() => {
+    playlist.shuffle();
+    setShuffle(true);
+    setTimeout(() => {
+      setShuffle(false);
+    }, 100);
+  }, [playlist]);
+
+  const currentTimeText = useMemo(() => {
+    const minutes = Math.floor(currentTime / 60);
```

Comment:
```suggestion
    const minutes = ~~(currentTime / 60);
```

양수에선 `Math.floor`와 `~~`가 동일한데 `~~`가 더빠르니까 바꿔주세요

Reply:
JS는 대체 어떤 언어일까